neat_core.c

```diff
@@ -804,6 +1035,210 @@ neat_open(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port)
     return neat_he_lookup(mgr, flow, he_connected_cb);
 }
 
+neat_error_code
+neat_open_localname(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char *localname)
+{
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    if (flow->name) {
+        return NEAT_ERROR_BAD_ARGUMENT;
+    }
+
+    if (localname == NULL)
+	return neat_open(mgr, flow, name, port);
+
+    return NEAT_ERROR_UNABLE;
+}
+
+neat_error_code
+neat_open_multistream(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char* localname, int count)
```

Comment:
I have reverted the commits that added these functions for now, until we have decided how to handle optional arguments.